src/contract_wrappers/modules/checkpoint/voting/advanced_p_l_c_r_voting_checkpoint_wrapper/common.ts

```diff
@@ -908,23 +935,30 @@ export default abstract class AdvancedPLCRVotingCheckpointCommon extends ModuleC
       return e.args;
     });
 
-    const onlyBallot = _.filter(args, e => {
+    const onlyBallot = _.find(args, e => {
       /* eslint-disable no-underscore-dangle */
       return e._ballotId.toNumber() === params.ballotId;
     });
+    assert.assert(onlyBallot !== undefined, ErrorCode.NotFound, 'Ballot id not found');
 
     const keccakKeys = ['uint256'];
-    params.choices.forEach(() => {
+    const bgVotes = params.choices.map(e => {
```

Comment:
Callbacks passed to functions like `map` should ideally never have side effects (like altering the `keccakKeys` array by pushing stuff to it). This is because (by convention), `map` is considered to be a "pure" function.

Normally I'd suggest using `forEach`, but since the `keccakKeys` array has the same element repeated n+1 times (where n is the length of bgVotes), you can use some lodash magic to make this easier by doing

```
const bgVotes = params.choices.map(e => valueToWei(new BigNumber(e), FULL_DECIMALS).toString());
const keccakKeys = times(bgVotes.length + 1, constant('uint256'));
```

(`times` and `constant` should be imported from lodash)